pkg/machinery/client/events.go

```diff
@@ -123,6 +123,84 @@ func (c *Client) EventsWatch(ctx context.Context, watchFunc func(<-chan Event),
 	}
 }
 
+// EventResult is the result of an event watch, containing either an Event or an error.
+type EventResult struct {
+	// Event is the event that was received.
+	Event Event
+	// Err is the error that occurred.
+	Error error
+}
+
+// EventsWatchV2 watches events of a single node and wraps the Events by providing a simpler interface.
+// It blocks until the first (empty) event is received, then spawns a goroutine that sends events to the given channel.
+// EventResult objects sent into the channel contain either the errors or the received events.
+//
+//nolint:gocyclo
+func (c *Client) EventsWatchV2(ctx context.Context, node string, ch chan<- EventResult, opts ...EventsOptionFunc) error {
```

Comment:
we probably don't want a `node` here, it should be part of the context

Reply:
Oh right, forgot that

Reply:
done